python/samples/concepts/agents/step1_agent.py

```diff
@@ -0,0 +1,63 @@
+# Copyright (c) Microsoft. All rights reserved.
+
+import asyncio
+
+from semantic_kernel.agents.chat_completion_agent import ChatCompletionAgent
+from semantic_kernel.connectors.ai.open_ai import AzureChatCompletion
+from semantic_kernel.contents.chat_history import ChatHistory
+from semantic_kernel.contents.utils.author_role import AuthorRole
+from semantic_kernel.kernel import Kernel
+
+###################################################################
+# The following sample demonstrates how to create a simple,       #
+# non-group agent that repeats the user message in the voice      #
+# of a pirate and then ends with a parrot sound.                  #
+###################################################################
+
+# To toggle streaming or non-streaming mode, change the following boolean
+streaming = True
+
+# Define the agent name and instructions
+PARROT_NAME = "Parrot"
+PARROT_INSTRUCTIONS = "Repeat the user message in the voice of a pirate and then end with a parrot sound."
+
+
+async def invoke_agent(agent: ChatCompletionAgent, input: str, kernel: Kernel, chat: ChatHistory):
+    """Invoke the agent with the user input."""
+    chat.add_user_message(input)
+
+    print(f"# {AuthorRole.USER}: '{input}'")
+
+    if streaming:
+        contents = []
+        content_name = ""
+        async for content in agent.invoke_streaming(kernel, chat):
+            content_name = content.name
```

Comment:
won't this be overwritten?

Reply:
It's the same name for the entire agent run, so not a big deal.